### ISSUE TITLE
Why is the prob.xval randomly constructed?

### ISSUE THREAD

**Issue by hnyz979** [mborgerding/onsager_deep_learning#4]
Dear Mark
I have another question about the ground truth sparse coding used to train the network. I find the loss is computed by tf.nn.l2_loss(xhat_ - prob.x_) (Line 60 in train.py), thus, the prob.x_ should be the ground truth. Then, I find that the prob.x_ is fed to be prob.xval (Line 107 in train.py), however, it seems that the prob.xval is actually randomly constructed (Line 50 in problem.py). 

In my understanding, the ground truth sparse coding should be given in a more "formal" way, so, I have two questions: (1) why can the ground truth sparse coding be actually randomly constructed in the code? (2) why don't we use the reconstruction loss \bold{y}-bold{A}\hat{\bold{x}}_T as the objective function? 

Thank you very much!
